src/nvim/tui/tui.c

```diff
@@ -1576,7 +1576,7 @@ void tui_suspend(TUIData *tui)
   tui_terminal_stop(tui);
   stream_set_blocking(tui->input.in_fd, true);   // normalize stream (#2598)
 
-  kill(0, SIGTSTP);
+  kill(0, SIGSTOP);  // Avoid SIGTSTP handler in os/signal.c. #33258
```

Comment:
without `SIGTSTP`, the `autowrite_all()` won't be called, which defeats the purpose of your other change.
```suggestion
  kill(0, SIGTSTP);  // Triggers autowrite_all() in os/signal.c.
  kill(0, SIGSTOP);
```

Reply:
Weirdly though, I ran the test `test/functional/autocmd/signal_spec.lua` (which contains checks of autowrite) with `kill(0, SIGTSTP)` removed, and it seems to be totally fine... Anyway, let's see the test.

Reply:
Oops. It seems the author pushed the change very quickly. It might be better to see the test result and then comment on it. Anyway I tested a lot on it and it *seems* adding or deleting the `kill(0, SIGTSTP)` doesn't affect anything though, including the functional tests (at least on my machine).

...also begging that ubuntu's tests don't have weird unrelated errors.

Reply:
> `ex_stop` has already a `auto_writeall` in it:

ok, makes sense. So we can drop the `kill(0, SIGTSTP);` in tui.c.

Reply:
> Another thing: `kill -TSTP client_pid` now no longer ~detach~"stop" the ui. But it will before [4dabeff](https://github.com/neovim/neovim/commit/4dabeff308222307ede8e74a2bd341713a7f7d81).

I think that's quite fair since we don't stop the UI in the signal handler, we only did that in UI's event handler. We may need to call some function on such signal to stop the UI as if the user is stopping it, maybe.